docs/CONTRIBUTING.md

```diff
@@ -0,0 +1,246 @@
+   for undocumented code
+5. Find a section of under-tested code and add some unit tests
+6. Take a look through the GitHub repo's [Issues](https://github.com/softwareconstruction240/autograder/issues) page
+   and find one you like, especially (but not limited to) ones labeled
+   ["good first issue"](https://github.com/softwareconstruction240/autograder/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22)
+
+Don't be afraid to submit a PR, and most importantly, just get sucked in!
+
+## Development Pipeline
+
+As a broad overview, all changes to the Autograder start as an issue. Each 
+issue eventually gets a branch made to solve the issue. And each branch
+gets made in
+
+### 1. Issues
+
+All changes to the Autograder start as an issue. Issues help us track what 
+needs to be done and who's working on what. You can pick an already 
+existing issue to work on, or you can write your own. 
+
+> [!TIP]
+> Ideally, you should first work on issues in the `On Deck` column of the 
+[TA Project Board](https://github.com/orgs/softwareconstruction240/projects/1/views/9),
+but you are welcome to pick any unassigned issue
+
+#### Creating an Issue
+- Create the issue from the TA Projects Board (linked above)
+  - Click "add item" at the bottom of the `Todo` list
+  - Type `#autograder` to connect the issue to the Autograder
+  - Type a title and hit enter
+- Use a clear, descriptive title that uses one of these prefixes:
+  - `Backend:` server side changes
+  - `Frontend:` client (web) side changes
+  - `Fullstack:` for issues that require front and backend changes
+  - `Docs:` changes to documentation (either markdown files or JSDocs/JavaDocs)
+  - `Tests:` adding or modifying tests
+  - `Dev:` changes to enhance the Autograder development experience
+- Provide enough context so others can understand the purpose
+- Add labels as appropriate ([Here's the list](https://github.com/softwareconstruction240/autograder/issues/labels))
+- Link related issues
+- Issues should be narrow in scope
+
+#### Working on an Issue
+- Assign the issue to yourself before beginning work (this is how we know 
+  who is working on what, and gives you a sort of ownership over the issue)
+- Create a new branch ([more info here](#2-branches))
+- If you find a new issue, file it separately (don't mix concerns)
+- Leave a comment from time to time with updates on your progress
+
+### 2. Branches
+Branches in git allow us to all work on separate things in the same codebase.
+You will do all your work in a branch.
+
+#### Creating a branch
+Create the branch directly from the issue so your work is linked.
+
+- Always branch from `main` 
+  - (exceptions exist if you know what you're doing)
+- Use descriptive names that reflect what you're working on:
+    - `add-late-submission`
+    - `admin-display-verification-status`
+    - `server-communicator`
+    - `extract-service-logic-from-controllers`
+- Use kebab-case (lowercase with hyphens)
+- Keep names concise but clear
```

Comment:
I think it would be useful to allow the naming of branches by GitHub--The `Create a branch` button on the issue page

Reply:
I agree that it's good to use issue numbers, but we also occasionally create PRs to address topics without creating an "Issue." I don't see a purpose in _requiring_ that an issue be created for every branch. The PR that will be generated from a branch is provides enough context. 

If you need help tracking stale branches, try the following resources:
* [GitHub Branches Page](https://github.com/softwareconstruction240/autograder/branches) lists all branches, and identifies them as "stale" or "active"
* `git log --oneline --graph --all --first-parent` can be used to visualize the relationships between many branches simultaneously. Scroll down the first page or so to start seeing the staler branches.
* We've enabled the feature to "automatically delete a branch when its PR is merged." This keeps the total number of branches down to _only_ those which are being worked on and those that never got finished.